src/CategoricalArrays.jl

```diff
@@ -36,4 +34,74 @@ module CategoricalArrays
     include("recode.jl")
 
     include("deprecated.jl")
+
+    function __init__()
+        @require JSON="682c06a0-de6a-54ab-a142-c8b1cf79cde6" begin
+            #import JSON
+            # JSON of CategoricalValue is JSON of the value it refers to
+            JSON.lower(x::CategoricalValue) = JSON.lower(unwrap(x))
+            DataAPI.defaultarray(::Type{CategoricalValue{T, R}}, N) where {T, R} =
+                CategoricalArray{T, N, R}
+            DataAPI.defaultarray(::Type{Union{CategoricalValue{T, R}, Missing}}, N) where {T, R} =
+                CategoricalArray{Union{T, Missing}, N, R}
+        end
+
+        @require RecipesBase="3cdcf5f2-1ef4-517c-9805-6587b60abb01" @eval begin
```

Comment:
it would be good to cover this with tests if it is not overly complex.

Reply:
It's tested AFAICT. Or do you mean something else?

Reply:
Just coveage test was signaling it is not tested. That is why I asked.

Reply:
You mean line 50 at https://github.com/JuliaData/CategoricalArrays.jl/runs/3633298660? That's probably a bug, as other lines in the block appear to be covered. I guess a macro within two macro calls is too complex for coverage checks.

Reply:
ok